src/window/Window.cpp

```diff
@@ -67,6 +67,24 @@ std::shared_ptr<std::vector<WindowBackend>> Window::GetAvailableWindowBackends()
     return mAvailableWindowBackends;
 }
 
+bool Window::IsAvailableWindowBackend(int32_t backendId) {
+    auto backend = static_cast<WindowBackend>(backendId);
+
+    // Verify the id is a valid backend enum value
+    switch (backend) {
+        case WindowBackend::FAST3D_DXGI_DX11:
+        case WindowBackend::FAST3D_SDL_OPENGL:
+        case WindowBackend::FAST3D_SDL_METAL:
+            break;
+        default:
+            return false;
+    }
```

Comment:
we can't just check to see if the value is `>= 0` and `< max` if we're setting values that aren't sequential, we could probably do some fancy template stuff or some fancy macro stuff instead of just throwing the values into a switch here but all of those options feel a bit overengineered for a use case this simple